src/pushsource/_impl/backend/koji_source.py

```diff
@@ -569,6 +570,9 @@ def _push_items_from_vmi_build(self, nvr, meta):
             extra = archive.get("extra") or {}
             image = extra.get("image") or {}
             arch = image.get("arch")
+            boot_mode = image.get("boot_mode")
+            if boot_mode:
+                BootMode(boot_mode)
```

Comment:
You create a BootMode but never assign it to anything, is that intentional?

The only reason I can think to do this is if you were trying to make it crash early if boot_mode was a value other than one of the allowed in the enum.  But given that you end up passing just the plain `boot_mode` str into the push item later, and there is no conversion into a `BootMode` there, I'm thinking it's more likely this is just a bug.

See also my later comment on the test which may explain why the test apparently passes despite this bug.

Reply:
Thanks! It's fixed now, PTAL again